java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Field.java

```diff
@@ -37,6 +40,7 @@
   private final String name;
   private final boolean nullable;
   private final ArrowType type;
+  private final Long dictionary;
```

Comment:
I may also be a bit confused -- if these Field objects are only used for record batch disassembly and reassembly, then these changes are fine